### PR TITLE
More groups tests

### PR DESCRIPTION
**Pull request by adfost**

### CODE REVIEW

File: integration/groups_test.go [hunk 1]
```diff
@@ -103,6 +103,208 @@ func Test_AddGroupAPI(t *testing.T) {
...
+	tests := []struct {
+		name           string
+		args           args
+		expectedStatus int
+		expectedError  error
+	}{
+		{
+			name: "Get Group - Valid",
+			args: args{
+				api: "/group?name=getgroup1",
```
Comment (dvaldivia) [inline integration/groups_test.go:124]: all these tests are meant for the same endpoint `/group` so I don't think that should be an argument, on the other hand `name` is a valid argument

**Review by dvaldivia — CHANGES_REQUESTED** [review]

File: integration/groups_test.go [hunk 2]
```diff
@@ -103,6 +103,208 @@ func Test_AddGroupAPI(t *testing.T) {
...
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+
+			client := &http.Client{
+				Timeout: 3 * time.Second,
+			}
+
+			// Add policy
```
Comment (dvaldivia) [inline integration/groups_test.go:277]: invalid comment

File: integration/groups_test.go [hunk 3]
```diff
@@ -103,6 +103,208 @@ func Test_AddGroupAPI(t *testing.T) {
...
```
Comment (dvaldivia) [inline integration/groups_test.go:203]: invalid comment

File: integration/groups_test.go [hunk 4]
```diff
@@ -103,6 +103,208 @@ func Test_AddGroupAPI(t *testing.T) {
...
```
Comment (dvaldivia) [inline integration/groups_test.go:146]: invalid comment

**Review by dvaldivia — CHANGES_REQUESTED** [review]

File: integration/groups_test.go [hunk 5]
```diff
@@ -145,14 +258,73 @@ func Test_DeleteGroupAPI(t *testing.T) {
...
+				Timeout: 3 * time.Second,
+			}
 
 			requestDataPolicy := map[string]interface{}{}
 
 			requestDataJSON, _ := json.Marshal(requestDataPolicy)
 			requestDataBody := bytes.NewReader(requestDataJSON)
 			request, err := http.NewRequest(
-				tt.verb, fmt.Sprintf("http://localhost:9090/api/v1%s", tt.args.api), requestDataBody)
+				tt.verb, "http://localhost:9090/api/v1/group?name=grouptests1", requestDataBody)
```
Comment (dvaldivia) [inline integration/groups_test.go:327]: please make the `name` an argument so you can also test attempting to delete an invalid group

**Review by dvaldivia — CHANGES_REQUESTED** [review]

File: integration/groups_test.go [hunk 6]
```diff
@@ -120,19 +304,28 @@ func Test_DeleteGroupAPI(t *testing.T) {
...
 			expectedStatus: 204,
 			expectedError:  nil,
 		},
+		{
+			name: "Delete Group - Invalid",
+			args: args{
+				api: "?name=grouptests12345",
+			},
+			verb:           "DELETE",
+			expectedStatus: 500,
```
Comment (dvaldivia) [inline integration/groups_test.go:320]: Attempting to deleting a non-existing resource should return 404, I see the API returns 
```
{"code":500,"detailedMessage":"The specified group does not exist. (Specified group does not exist)","message":"an error occurred, please try again"}
```

Can you update the API tor return 404 for this specific error?

**Review by dvaldivia — CHANGES_REQUESTED** [review]

File: restapi/admin_groups.go [hunk 7]
```diff
@@ -204,9 +204,12 @@ func getRemoveGroupResponse(session *models.Principal, params groupApi.RemoveGro
 	// createad a MinIO Admin Client interface implementation
 	// defining the client to be used
 	adminClient := AdminClient{Client: mAdmin}
-
 	if err := removeGroup(ctx, adminClient, params.Name); err != nil {
-		return ErrorWithContext(ctx, err)
+		err2 := ErrorWithContext(ctx, err)
```
Comment (dvaldivia) [inline restapi/admin_groups.go:208]: Parse the error from MinIO via `ToErrorResponse` and check the code alone
```
minioError := madmin.ToErrorResponse(err)
		if minioError.Code == "XMinioAdminNoSuchGroup" {
			
		}
```

**Review by dvaldivia — CHANGES_REQUESTED** [review]

**Review by dvaldivia — APPROVED** [review]

**Review by bexsoft — APPROVED** [review]

**Pull request merged**